docs/spec/v1beta2/helmrepositories.md

```diff
@@ -162,6 +162,133 @@ A HelmRepository also needs a
 
 Possible values are `default` for a Helm HTTP/S repository, or `oci` for an OCI Helm repository.
 
+
+### Provider
+
+`.spec.provider` is an optional field that allows specifying an OIDC provider used
+for authentication purposes.
+
+Supported options are:
+- generic
+- aws
+- azure
+- gcp
+
+The generic provider can be used for public repositories or when static credentials
+are used for authentication. If you do not specify `.spec.provider`, it defaults
+to generic.
+
+**Note**: The provider field is supported only for Helm OCI repositories. The `spec.type`
+field must be set to `oci`.
+
+#### AWS
+
+The aws provider can be used to authenticate automatically using the EKS worker
```

Comment:
```suggestion
The `aws` provider can be used to authenticate automatically using the EKS worker
```

A lot of the paragraphs below are missing these back ticks. Please refer to their usage in OCIRepository spec docs.